internal/dag/gatewayapi_processor.go

```diff
@@ -61,6 +63,41 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 	for _, listener := range p.source.gateway.Spec.Listeners {
 
+		var listenerSecret *Secret
+		var err error
+
+		// Check for TLS on the Gateway.
+		if tls := listener.TLS; tls != nil {
+			// Validate the CertificateRef is configured.
+			if tls.CertificateRef == nil {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "SecretNotConfigured",
+					"Spec.VirtualHost.TLS Secret is not configured")
+				continue
+			}
+
+			// Validate the correct protocol is specified.
+			if listener.Protocol != gatewayapi_v1alpha1.HTTPSProtocolType {
+				p.Error(contour_api_v1.ConditionTypeTLSError, "InvalidProtocol",
+					"Spec.VirtualHost.Protocol %q is not valid.")
+				continue
+			}
+
+			// Validate a v1.Secret is referenced which can be kind: secret & group: core.
+			// ref: https://github.com/kubernetes-sigs/gateway-api/pull/562
+			if strings.ToLower(tls.CertificateRef.Kind) != "secret" && strings.ToLower(tls.CertificateRef.Group) != "v1" {
```

Comment:
Should be an `OR`, otherwise e.g. `v1/configmap` would be allowed.
```suggestion
			if strings.ToLower(tls.CertificateRef.Kind) != "secret" || strings.ToLower(tls.CertificateRef.Group) != "v1" {
```

Might actually be easiest to just define a positive-polarity `isSecretRef` func and then check the negation of that:
```
func isSecretRef(certificateRef) bool {
        return certificateRef.Group == "v1" && certificateRef.Kind == "secret"
}
...
if !isSecretRef(tls.CertificateRef) {
        ....
}
```

Reply:
> Should be an OR, otherwise e.g. v1/configmap would be allowed.

Wouldn't the `OR` allow the configmap to work? I'm confused on that bit.

Reply:
The current code says that to be an invalid ref, it has to both (a) not be a `Secret`; **and** (b) not be in `v1`. Since `v1/ConfigMap` doesn't meet condition (b), it wouldn't be considered invalid. But really, if it meets either (a) **or** (b) it should be invalid.

I think just doing the `isSecretRef` helper, and then checking the negation of it, is easiest :)